Doc/Manual/Fortran.html

```diff
@@ -688,105 +652,119 @@ <H3><a name="Fortran_byte_strings">Byte strings</a></H3>
+Some C++ enumeration definitions cannot be natively interpreted by a Fortran compiler (e.g. <code>FOO = 0x12,</code> or <code>BAR = sizeof(int),</code>), so these are defined in the C++ wrapper code and <em>bound</em> in the Fortran wrapper code:
 </p>
 
 
-<p>
-Unlike many other SWIG target languages, the Fortran-wrapped classes are <em>strongly typed</em>: the compiler enforces type checking between data types and function arguments.
-</p>
+<div class="targetlang"><pre><code>integer(C_INT), protected, public, &amp;
+   bind(C, name=&quot;swigc_FOO&quot;) :: FOO</code></pre>
+</div>
 
-<H3><a name="Fortran_ignored_or_unimplemented_forward_declared_classes">Ignored or unimplemented forward-declared classes</a></H3>
 
 <p>
-Some functions may include references or pointers to classes that are not wrapped by Fortran proxy functions. In these cases, an opaque derived type called <code>SwigUnknownClass</code> will be generated and used as a placeholder for the argument or return value. These could theoretically be passed between wrapped SWIG functions, although no type checking will be performed to ensure that the unknown classes are the correct types.
+The <code>%enumerator</code> and <code>%noenumerator</code> directives can be used to explicitly enable and disable treatment of a C++ <code>enum</code> as a Fortran enumerator. Disabling the enumerator feature causes the value to be wrapped as externally-bound C integers.
 </p>
 
-<H3><a name="Fortran_enumerations">Enumerations</a></H3>
 
 <p>
-Fortran 2003 implements C enumerations using the <code>ENUM, BIND(C)</code> statement. These enumerators are simply a set of loosely grouped compile-time integer constants that are guaranteed to be compatible with C enumerators. Unlike C++, all enumerators in Fortran are anonymous.
+If an enumeration type has not been defined but is used in a function signature, a placeholder <code>SwigUnknownEnum</code> enumerator will be generated and used instead.
 </p>
 
 
 <p>
-To associate a C enumeration name with the Fortran generated wrappers, SWIG generates an additional enumeration with the C class name and a dummy value of <code>-1</code>. The enumeration generated from the C code
+Class-scoped enumerations are prefixed with the class name:
 </p>
 
 
-<div class="code"><pre><code>enum MyEnum {
-    RED = 0,
-    GREEN,
-    BLUE,
-    BLACK = -1
-};
-</code></pre>
+<div class="code"><pre><code>struct MyStruct {
+  enum Foo {
+    Bar = 0
+  };
+};</code></pre>
 </div>
 
 
 <p>
-looks like:
+generates
 </p>
 
 
-<div class="targetlang"><pre><code> enum, bind(c)
-  enumerator :: MyEnum = -1
-  enumerator :: RED = 0
-  enumerator :: GREEN = RED + 1
-  enumerator :: BLUE = GREEN + 1
-  enumerator :: BLACK = -1
- end enum
-</code></pre>
+<div class="targetlang"><pre><code>enum, bind(c)
+ enumerator :: MyStruct_Foo = -1
```

Comment:
Why is there an additional enum value with the value of -1?

Reply:
Without this extra value, the enumeration type can only be referenced by one of its values -- e.g.  instead of `integer(kind(Color))` the wrapper code would have to be `integer(kind(RED))`.

The `= -1` value was just a way of tagging the enumeration, although I can definitely see how it could be found confusing, clunky, and nonstandard. Do you think the usability would be improved by eliminating this behavior?

Reply:
I've actually come up with a better solution by simply defining  a consistent integer type outside of the enumeration itself. See https://github.com/sethrj/swig/pull/90 .